src/core/document.js

```diff
@@ -308,29 +307,112 @@ var Page = (function PageClosure() {
+            annotationsData.push(annotations[i].data);
+          }
         }
-      }
-      return annotationsData;
+        return annotationsData;
+      });
     },
 
     get annotations() {
-      var annotations = [];
+      var AnnotationWorkerTask = (function AnnotationWorkerTaskClosure() {
+        function AnnotationWorkerTask(name) {
+          this.name = name;
+          this.terminated = false;
+          this.capability = createPromiseCapability();
+        }
+
+        AnnotationWorkerTask.prototype = {
+          get finished() {
+            return this.capability.promise;
+          },
+
+          finish() {
+            this.capability.resolve();
+          },
+
+          terminate() {
+            this.terminated = true;
+          },
+
+          ensureNotTerminated() {
+            if (this.terminated) {
+              throw new Error('Annotation worker task was terminated');
+            }
+          },
+        };
+
+        return AnnotationWorkerTask;
+      })();
+
+      // create a blank annotation fonts array
+      // if it's not initialized yet
+      if (this.pdfManager && this.pdfManager.pdfDocument &&
+          this.pdfManager.pdfDocument.acroForm &&
+          !this.pdfManager.pdfDocument.acroForm.annotationFonts) {
+        this.pdfManager.pdfDocument.acroForm.annotationFonts = [];
+      }
+
+      var task = new AnnotationWorkerTask(
+        'GetAnnotationAppereances: page ' + this.pageIndex);
+
+      var handler = {};
+
+      var self = this;
+
+      handler.send = function (actionname, data) {
+        if (self.pdfManager && self.pdfManager.pdfDocument &&
+            self.pdfManager.pdfDocument.acroForm) {
+          self.pdfManager.pdfDocument.acroForm.annotationFonts.push(data);
+        }
+      };
+
+      var pageNum = this.pageIndex + 1;
+      var start = Date.now();
+
+      var partialEvaluator = new PartialEvaluator({
+        pdfManager: this.pdfManager,
+        xref: this.xref,
+        handler,
+        pageIndex: this.pageIndex,
+        idFactory: this.idFactory,
+        fontCache: this.fontCache,
+        builtInCMapCache: this.builtInCMapCache,
+        options: this.evaluatorOptions,
+      });
+
       var annotationRefs = this.getInheritedPageProp('Annots') || [];
+      var annotationsPromise = [];
       for (var i = 0, n = annotationRefs.length; i < n; ++i) {
         var annotationRef = annotationRefs[i];
-        var annotation = AnnotationFactory.create(this.xref, annotationRef,
-                                                  this.pdfManager,
-                                                  this.idFactory);
-        if (annotation) {
-          annotations.push(annotation);
-        }
+          annotationsPromise.push(AnnotationFactory.create(
+          this.xref,
+          annotationRef,
+          this.pdfManager,
+          this.idFactory,
+          partialEvaluator,
+          task
```

Comment:
End with a trailing comma.

Reply:
I'm not sure for that, this is function call, do we need this comma after task (if you meant that)

Reply:
Yes, a comma after `task`. The reason to end with a trailing comma is that if in the future we add more parameters, you don't need to touch this line again. It makes it a bit easier in the Git logs.

Reply:
I'm still a little bit confused. Here is the function call, if I add the comma after task Lint generates an error.

static create(xref, ref, pdfManager, idFactory, evaluator, task);

var annotationPromise = AnnotationFactory.create(
          this.xref,
          annotationRef,
          this.pdfManager,
          this.idFactory,
          partialEvaluator,
          task
        );

Here is the current code from master, there is no command after the last parameter as well:

var annotation = AnnotationFactory.create(this.xref, annotationRef,
                                                  this.pdfManager,
                                                  this.idFactory);

Reply:
I thought this was an object and did not see it was a function call. You can disregard my comment; sorry for the noise!